### PR TITLE
Add tags

### PR DESCRIPTION
**Pull request by szczys**
Add tags to Project and Device

### CODE REVIEW

File: golioth/cli.py [hunk 1]
```diff
@@ -617,6 +617,32 @@ async def delete(config, name):
 
         await project.delete_device(name)
 
+@device.command()
+@click.argument('name')
+@click.argument('tag_id')
+@pass_config
+async def tag_add(config, name, tag_id):
```
Comment (sam-golioth) [inline golioth/cli.py:624]:
```suggestion
async def add_tag(config, name, tag_id):
```
Python tends to use verb-noun construction for function names. And this function name translates directly into CLI command, where verb-noun is also more common.

File: golioth/cli.py [hunk 2]
```diff
@@ -617,6 +617,32 @@ async def delete(config, name):
...
+        project = await Project.get_by_id(client, config.default_project)
+        device = await project.device_by_name(name)
+
+        await device.tag_add(tag_id)
+
+@device.command()
+@click.argument('name')
+@click.argument('tag_id')
+@pass_config
+async def tag_remove(config, name, tag_id):
```
Comment (sam-golioth) [inline golioth/cli.py:637]:
```suggestion
async def remove_tag(config, name, tag_id):
```

File: golioth/cli.py [hunk 3]
```diff
@@ -867,6 +867,80 @@ async def delete(config, device_name, key):
...
+@pass_config
+async def get_id(config, tag_name):
+    """Get Tag ID by name"""
+    with console.status(f'Getting Tag ID for {tag_name}...'):
+        client = Client(api_url = config.api_url, api_key = config.api_key, access_token = config.access_token)
+        project = await Project.get_by_id(client, config.default_project)
+
+        b_id = await project.tags.get_id(tag_name)
+
+        console.print(b_id)
```
Comment (szczys) [inline golioth/cli.py:937]: New since last review

File: golioth/golioth.py [hunk 4]
```diff
@@ -881,3 +882,85 @@ async def delete(self, key: str):
...
+    async def get(self, tag_id: str):
+        resp = await super().get(tag_id)
+        return Tag(self.project, resp.json()['data'])
+
+    async def get_id(self, tag_name: str):
+        tag_list = await self.get_all()
+        for t in tag_list:
+            if t.name == tag_name:
+                return t.id
+        return None
```
Comment (szczys) [inline golioth/golioth.py:971]: New since last review.

**Review by szczys — COMMENTED** [review]
Thanks for the pointers, I've made all of the updates.

I also added on additional function `get_id()` which returns the Tag ID for the supplied Tag name (or None)

**Review by sam-golioth — APPROVED** [review]

**Pull request merged**